build/Version.props

```diff
@@ -0,0 +1,124 @@
+	
+    <Authors>Microsoft</Authors>
+    <ProjectUrl>https://github.com/aspnet/websdk</ProjectUrl>
+    <LicenseUrl>https://www.microsoft.com/web/webpi/eula/net_library_eula_enu.htm</LicenseUrl>
+    <ImageUrl>http://go.microsoft.com/fwlink/?LinkID=288859</ImageUrl>
+    <Tags>WebSdk</Tags>
+    <Company>Microsoft</Company>
+    <Copyright>Copyright © Microsoft Corporation</Copyright>	
+    <RequireLicenseAcceptance>true</RequireLicenseAcceptance>
+	
+    <GenerateAssemblyInfo Condition="'$(BuildingInsideVisualStudio)' != ''">False</GenerateAssemblyInfo>
+    <GenerateAssemblyFileVersionAttribute>False</GenerateAssemblyFileVersionAttribute>
+    <GenerateAssemblyVersionAttribute>False</GenerateAssemblyVersionAttribute>
+    <GenerateAssemblyInformationalVersionAttribute>False</GenerateAssemblyInformationalVersionAttribute>
+  </PropertyGroup>
+  
+  <PropertyGroup>
+    <AssemblyOriginatorKeyFile>$(WebSdkRoot)\tools\Key.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <DelaySign>true</DelaySign>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ArtifactsDir>$(WebSdkRoot)\artifacts\</ArtifactsDir>
+    <GitInfoFile>$(ArtifactsDir)GitInfo.cs</GitInfoFile>
+    <GitInfoProps>$(ArtifactsDir)GitInfo.props</GitInfoProps>
+  </PropertyGroup>
+  
+  <Target Name="CollectGitInfo">
+    <PropertyGroup>
+      <GitInfoCommitCount Condition="'$(COMMIT_HASH)' != ''">$(COMMIT_COUNT)</GitInfoCommitCount>
+      <GitInfoCommitHash Condition="'$(COMMIT_HASH)' != ''">$(COMMIT_HASH)</GitInfoCommitHash>
+      <GitExitCode>0</GitExitCode>
+    </PropertyGroup>
+
+    <Exec Command="git rev-list --count HEAD" ConsoleToMSBuild="true" Condition="'$(COMMIT_HASH)' == ''" IgnoreExitCode="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitCount" />
+      <Output TaskParameter="ExitCode" PropertyName="GitExitCode" />
+    </Exec>
+
+    <Exec Command="git rev-parse HEAD" ConsoleToMSBuild="true" Condition="'$(COMMIT_HASH)' == ''" IgnoreExitCode="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitInfoCommitHash" />
+      <Output TaskParameter="ExitCode" PropertyName="GitExitCode" />
+    </Exec>
+
+    <PropertyGroup Condition="'$(GitExitCode)' != '0'">
+      <GitInfoCommitCount>0</GitInfoCommitCount>
+      <GitInfoCommitHash>UNKNOWN</GitInfoCommitHash>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <GitCommitInfoContent>
+<![CDATA[
+[assembly: System.Reflection.AssemblyVersion("$(AssemblyReleaseVersion)")]
+[assembly: System.Reflection.AssemblyFileVersion("$(AssemblyReleaseVersion)")]
+[assembly: System.Reflection.AssemblyInformationalVersion("$(PackageVersion)+$(GitInfoCommitCount).$(GitInfoCommitHash)")]
+
+namespace Microsoft.WebSdk
+{
+    internal static class GitInfo
```

Comment:
Why not check this file in and have it read attributes instead?

Reply:
Added a checked-in version for fallback. I think it is a lot easier to understand what values are written if it is closer to Version.props